src/coreComponents/mesh/FaceManager.cpp

```diff
@@ -114,11 +114,14 @@ void FaceManager::setDomainBoundaryObjects()
 
 void FaceManager::setGeometricalRelations( CellBlockManagerABC const & cellBlockManager,
                                            ElementRegionManager const & elemRegionManager,
-                                           NodeManager const & nodeManager )
+                                           NodeManager const & nodeManager, bool baseMeshLevel )
 {
   GEOSX_MARK_FUNCTION;
 
-  resize( cellBlockManager.numFaces() );
+  if( baseMeshLevel )
```

Comment:
`if-based-architcture` will not scale in the whole GEOSX and will be complicated to maintain and to make evolve.

Reply:
Thank you for your comment. The reason why we have the "if (baseMeshLevel)" statement is to avoid redundant operations on higher-level meshes that have already been handled by the base-level mesh. For example, the "computeGeometry" function on line 136 is only needed to be called once for the base-level mesh.

We opted to use the "if-based-architecture" instead of creating new functions (that would include a reduced set of operations in "setGeometricalRelations" for higher-level meshes) in order to minimize code changes. As you can see, "setGeometricalRelations" is called by "nodeManager", "edgeManager", and "faceManager", which means we need to add new function to each of these classes.

Furthermore, we have to add additional code to "generateMeshLevel" to call different functions for the base-level mesh and higher-level meshes, which basically perform similar operations. There might be other options available, if you have a better way to handle it in mind, we welcome any suggestions for improving the code. :)